addon/components/bs-dropdown.js

```diff
@@ -171,6 +171,10 @@ export default Component.extend(ComponentParent, {
     } else {
       $(document).off(this.clickEventName);
     }
+
+    const eventToTrigger = (this.get('open') ? this.onShow : this.onHide) || function () {}
+
+    eventToTrigger()
```

Comment:
What about calling `openDropdown`/`closeDropdown` and `toggleDropdown` programatically? Shouldn't those also call `onShow` and `onHide`? I think those actions are yielded.

Reply:
Instead of dealing with those separate places; I put the event triggering directly to that ugly observer. I guess it will be fine like that; I covered all the cases hopefully by putting it into observer.

Reply:
Yup, I understand now, thx for explanation.